pkg/engine/artifacts_test.go

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+package engine
+
+import (
+	"testing"
+
+	"github.com/Azure/aks-engine/pkg/api"
+	"github.com/Azure/go-autorest/autorest/to"
+)
+
+func TestKubernetesContainerAddonSettingsInit(t *testing.T) {
```

Comment:
@CecileRobertMichon @jadarsie I'll finish out unit tests in here for all this functionality (including Azure Stack) before we merge this PR. The total lack of unit tests here probably explains why we weren't continuing to maintain this area as the proper interface for adding cluster configuration gates around particular addons.